### PR TITLE
feat: Support batch processing

### PR DESCRIPTION
**Pull request by KevinJBoyer**
## Ticket

https://navalabs.atlassian.net/browse/DST-580

## Changes
 - Add an interface for processing batch .csv files
 - Add pytest-asyncio to allow testing the async functions

## Context for reviewers
 - Processing must run async (or the Chainlit UI will timeout)
 - I will test this after merge with a non-engineer to confirm it also meets reqs for DST-581

## Testing
Uploading the test file suggested in the documentation:
image
The results:
image
Including citation details:
image

### CODE REVIEW

File: app/src/chainlit.py [hunk 1]
```diff
@@ -225,3 +243,29 @@ def _get_retrieval_metadata(result: OnMessageResult) -> dict:
...
+        author="backend",
+        content="Received file, processing...",
+    ).send()
+
+    try:
+        engine: chat_engine.ChatEngineInterface = cl.user_session.get("chat_engine")
+        result_file_path = await batch_process(file.path, engine)
+
+        # E.g., "abcd.csv" to "abcd_results.csv"
+        result_file_name = file.name.rsplit(".csv", 1)[0] + "_results.csv"
```
Comment (yoomlam) [inline app/src/chainlit.py:259]: `.removesuffix()` is available
```suggestion
        result_file_name = file.name.removesuffix(".csv") + "_results.csv"
```

File: app/src/chainlit.py [hunk 2]
```diff
@@ -225,3 +243,29 @@ def _get_retrieval_metadata(result: OnMessageResult) -> dict:
...
+
+async def _batch_proccessing(file: AskFileResponse) -> None:
+    await cl.Message(
+        author="backend",
+        content="Received file, processing...",
+    ).send()
+
+    try:
+        engine: chat_engine.ChatEngineInterface = cl.user_session.get("chat_engine")
+        result_file_path = await batch_process(file.path, engine)
```
Comment (yoomlam) [inline app/src/chainlit.py:256]: `batch_process()` returns the full path to the result file, including the filename, right? And it's being passed to `cl.File` on line 263 as the folder `path` -- interesting.

Comment (KevinJBoyer) [reply app/src/chainlit.py:256]: Yup, then cl.File exposes the (local file path) on a public URL like e.g., `chat/project/file/d1c04d6c-3439-4c0f-820f-cac977cb0c30?session_id=ed644005-9fdd-440e-936d-dee1eb7f87cb`

File: app/src/chainlit.py (hunk 1 above)
Comment (KevinJBoyer) [reply app/src/chainlit.py:259]: TIL!

**Pull request merged**